yaya_base/shiori3.dic

```diff
@@ -1953,9 +1953,28 @@ SHIORI3EV.On_enable_debug : void
 
 SHIORI3EV.On_ShioriEcho
 {
-	if SHIORI3FW.DebugMode {
+	if ISFUNC('On_ShioriEcho') {//User costom ShioriEcho
+		EVAL('On_ShioriEcho')
+	}
+	elseif SHIORI3FW.DebugMode {
 		'\0Eval Result = ' + JOIN(EVAL(reference[0]),',')
 	}
+	else {
+		'\0Need DebugMode'
+	}
+}
+
+SHIORI3EV.On_Has_Event : void {
+	_event_name=reference0
```

Comment:
It is better to use `reference[0]` instead of `reference0`.

Reply:
Agree with nikolat. The rest looks good.

Reply:
fixed at https://github.com/YAYA-shiori/yaya-dic/commit/5921e116d2bded8c9a750b602857c815d0caf1fb#diff-fb0011643823c1dc75efe4ec8109cd34ad0a6058d183913a7a3c94bba4817287
thanks for this bug catch!